rfd/0097-teleport-connect-usage-metrics.md

```diff
@@ -22,35 +22,65 @@ features, which ones are the most popular and which are problematic for users.
 
 ## Details
 
-Connect will have a stateless metrics service that will send events to an unauthorized gRPC endpoint, exposed by TSH
-daemon. The anonymization should be done in `tshd` (to be open on how it is done and what is sent further). Events will
-be sent to `prehog` directly.
-It was considered to use authorized endpoint provided by cluster's `Auth Service`, but it seems to not work well for
+### Collecting events
+
+Events for Connect will be collected on the client side. TypeScript code will have a stateless metrics service that will
+forward them to the gRPC handler exposed by `tsh daemon`. To prevent flooding backend with a large number of small
+requests, events will be batched before sending to `prehog`. The batching mechanism has already been implemented
+in `UsageReporter` that will be used for collecting cluster events. `tsh deamon` will try to reuse the same code as much
+as possible (by providing its own batching parameters and submit function). Events will be sent every once an hour (this
+may change) and before closing the app.
+
+It was considered to use an authorized endpoint provided by cluster's `Auth Server`, but it seems to not work well for
 Connect for a few reasons:
 
-- Some events do not belong to any cluster, like _app start_ event.
-- A user can use Connect while not logged in. For example, new tabs can be opened after the session expires.
+- Some events may not belong to any cluster (at the time of writing this RFD there is no such event, but the solution
+  should be future-proof).
+- Batch can contain events from multiple clusters.
+- Batch can be sent after the session expires.
+
+### Anonymization
+
+Each event that contains sensitive data, like cluster name needs to be anonymized. It will be done in `tsh deamon`, the
+same way as in `Auth Server` - using HMAC with `cluster id` as the key. Connect will reuse the same code.
+The only issue with anonymizing events client-side is lack of `cluster id` that is kept in `Auth Server`. To remedy
+this, when the app starts and retrieves cluster information, it should also retrieve the `cluster id`, create an
+anonymizer and store it in the cluster struct.
```

Comment:
What kind of cluster ID does this refer to?

Reply:
It is the unique cluster ID that is set once during the first auth server startup 
https://github.com/gravitational/teleport/blob/master/lib/services/clustername.go#L31